packages/yarnpkg-fslib/sources/ZipFS.ts

```diff
@@ -313,6 +315,10 @@ export class ZipFS extends BasePortableFakeFS {
     if (rc === -1)
       throw this.makeLibzipError(this.libzip.getError(this.zip));
 
+    // zip_close doesn't persist empty archives
+    if (this.entries.size === 0)
```

Comment:
Since we know it's empty we can just discard the zip archive to avoid unnecessary IO, we can also set the `mode` directly in the write to skip the `chmodSync` call